### PR TITLE
Add a way for consumers to use incremental build with Icon const enums

### PR DESCRIPTION
**Pull request by ThomasMichon**
# Overview

Presently, downstream projects which try to use incremental compilation with Webpack face an issue when using `@uifabric/icons` or `@uifabric/file-type-icons`: value references to `IconNames` or `FileIconType` will produce a runtime error, because the `transpileOnly` setting does not inline the `const enum` values, but they have been erased from the package `lib` output.

Thus consumers are faced with a dilemma: either disable `transpileOnly` and lose fast, incremental support, or work around the issue by manually in-lining values but losing type safety.

An alternate strategy exists, which is to keep the `const enum` exports as available value references, but export a companion type which represents the set of allowed values directly. By updating APIs to accept this new input type instead of the `const enum` type, they allow consumers to manually in-line the values while preserving type safety.

e.g.
```ts
const iconName: IconNames = 'Cancel'; // Error.
const iconName: IconNames = 'CancelFooBar' as IconNames; // Compiles, but permits any string!
const iconName: IconNamesInput = 'Cancel'; // Compiles.
```

This change adds an `IconNamesInput` type and a `FileIconTypeInput` type, which may be used everywhere they original types were used for their type reference, while preserving the original `const enum` entries to avoid breaking changes.

Nothing is added to the bundle size.

### CODE REVIEW

File: packages/icons/src/IconNames.ts [hunk 1]
```diff
@@ -1346,8 +1346,8 @@ export const enum IconNames {
   HalfCircle = 'HalfCircle',
   QuarterCircle = 'QuarterCircle',
   ThreeQuarterCircle = 'ThreeQuarterCircle',
-  SixPointStar = '6PointStar',
-  TwelvePointStar = '12PointStar',
+  SixPointStar = 'SixPointStar',
+  TwelvePointStar = 'TwelvePointStar',
```
Comment (Jahnp) [inline packages/icons/src/IconNames.ts]: This is a fair change, but just for context, the 6/12 names come from the icons team. That said, the icon tool can be updated to use these aliases and be more consistent. I'll do that today.

Comment (dzearing) [reply packages/icons/src/IconNames.ts]: should we add aliases for these?

File: packages/icons/src/IconNames.ts [hunk 2]
```diff
@@ -1495,3 +1495,5 @@ export const enum IconNames {
   OfficeChatSolid = 'OfficeChatSolid',
   MailSchedule = 'MailSchedule'
 }
+
+export type IconNamesInput = keyof typeof IconNames;
```
[the remark this answers is not in the document]
Comment (dzearing) [reply packages/icons/src/IconNames.ts]: can this be exported as "Icons" or "IconKeys"?

the word "Input" is misleading at best.

[the remark this answers is not in the document]
Comment (Jahnp) [reply packages/icons/src/IconNames.ts]: @dzearing good point. I'd be in favor of `IconKeys`--more self-explanatory than just `Icons`.

File: packages/file-type-icons/src/FileIconType.ts [hunk 3]
```diff
@@ -12,3 +12,5 @@ export const enum FileIconType {
   listItem = 4,
   sharedFolder = 5
 }
+
+export type FileIconTypeInput = 1 | 2 | 3 | 4 | 5;
```
Comment (cliffkoh) [inline packages/file-type-icons/src/FileIconType.ts]: nit: extra line on line 7 and missing jsdoc for this

File: packages/icons/src/IconNames.ts (hunk 2 above)
Comment (cliffkoh) [inline packages/icons/src/IconNames.ts]: jsdoc :)

**Pull request merged**